server/src/main/java/org/elasticsearch/script/field/ToScriptField.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.script.field;
+
+import org.apache.lucene.index.SortedNumericDocValues;
+import org.elasticsearch.index.fielddata.ScriptDocValues.Dates;
+import org.elasticsearch.index.fielddata.ScriptDocValues.Doubles;
+import org.elasticsearch.index.fielddata.ScriptDocValues.Longs;
+import org.elasticsearch.index.fielddata.SortedNumericDoubleValues;
+
+public interface ToScriptField<T> {
```

Comment:
Can you add some javadocs to give an idea of what this is used for?

Reply:
Done!

Reply:
Small comment, for me it'd be clearer to start with what it is (as opposed to the "why"). It's a helper that lets `LeafFieldData#getScriptField` translate from raw doc values to the field provider that the scripting API requires. It can make use of information specific to the field type, in a way `LeafFieldData` is not designed to do.